src/mono/System.Private.CoreLib/src/System/Threading/LowLevelLifoAsyncWaitSemaphore.Browser.Threads.Mono.cs

```diff
@@ -0,0 +1,226 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Threading;
+
+// <summary>
+// This class provides a way for browser threads to asynchronously wait for a semaphore
+// from JS, without using the threadpool.  It is used to implement threadpool workers.
+// </summary>
+internal sealed partial class LowLevelLifoAsyncWaitSemaphore : LowLevelLifoSemaphoreBase, IDisposable
+{
+    private IntPtr lifo_semaphore;
+
+    [MethodImplAttribute(MethodImplOptions.InternalCall)]
+    private static extern IntPtr InitInternal(int kind);
+
+    public LowLevelLifoAsyncWaitSemaphore(int initialSignalCount, int maximumSignalCount, int spinCount, Action onWait)
+        : base (initialSignalCount, maximumSignalCount, spinCount, onWait)
+    {
+        CreateAsyncWait(maximumSignalCount);
+    }
+
+#pragma warning disable IDE0060
+    private void CreateAsyncWait(int maximumSignalCount)
+#pragma warning restore IDE0060
+    {
+        lifo_semaphore = InitInternal(2);
```

Comment:
named constants for kind please

Reply:
Actually I think I'm going to make two separate icalls.   `InitInternal(int kind)` is a holdover from before I had two separate subclasses of semaphore.  There's no good reason for it.